frontend/web/components/metadata/AddMetadataToEntity.tsx

```diff
@@ -0,0 +1,251 @@
+            return (
+              item.field === meta.id && item.content_type === entityContentType
+            )
+          })
+          // Determine if isRequiredFor should be true or false based on is_required_for array
+          const isRequiredFor =
+            matchingItem &&
+            matchingItem.is_required_for &&
+            matchingItem.is_required_for.length > 0
+
+          // Return the metadata field with additional metadata model field information including isRequiredFor
+          return {
+            ...meta,
+            isRequiredFor: isRequiredFor || false,
+            metadataModelFieldId: matchingItem ? matchingItem.id : null,
+          }
+        })
+      if (projectFeatureData?.metadata && projectFeatureDataLoaded) {
+        const mergeLists = (
+          metadata: ProjectFlag['metadata'],
+          metadataField: CustomMetadataField[],
+        ) => {
+          const map = new Map(
+            metadataField.map((item) => [item.metadataModelFieldId, item]),
+          )
+          return metadataField.map((item) => ({
+            ...item,
+            ...(map.get(item.model_field!) || {}),
+            ...(metadata.find(
+              (m) => m.model_field === item.metadataModelFieldId,
+            ) || {}),
+          }))
+        }
+
+        const mergedList = mergeLists(
+          projectFeatureData?.metadata,
+          metadataForContentType,
+        )
+        const sortedArray = sortBy(mergedList, (m) =>
+          m.isRequiredFor ? -1 : 1,
+        )
+        setMetadataFieldsAssociatedtoEntity(sortedArray)
+      } else {
+        const sortedArray = sortBy(metadataForContentType, (m) =>
+          m.isRequiredFor ? -1 : 1,
+        )
+        setMetadataFieldsAssociatedtoEntity(sortedArray)
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    metadataFieldList,
+    metadataFieldListLoaded,
+    metadataModelFieldList,
+    metadataModelFieldListLoaded,
+    projectFeatureDataLoaded,
+    projectFeatureData,
+  ])
+  return (
+    <>
+      <FormGroup className='mt-4 setting'>
+        <PanelSearch
+          className='mt-1 no-pad'
+          header={
+            <Row className='table-header'>
+              <Flex className='table-column'>Metadata</Flex>
+              <Flex className='table-column'>Value</Flex>
+              {/* <Flex className='table-column'>Delete</Flex> */}
+            </Row>
+          }
+          items={metadataFieldsAssociatedtoEntity}
+          renderRow={(m: CustomMetadata) => {
+            return <MetadataRow metadata={m} isEdit={true} />
+          }}
+        />
+      </FormGroup>
+    </>
+  )
+}
+
+type MetadataRowType = {
+  metadata: CustomMetadata
+  onDelete?: () => void
+  isEdit: boolean
+}
+const MetadataRow: FC<MetadataRowType> = ({ isEdit, metadata }) => {
+  const [metadataValue, setMetadataValue] = useState<string>(
+    metadata?.field_value || '',
+  )
+  return (
```

Comment:
What does metadataValue do here? I can't see how this is submitted anywhere?

Reply:
I already corrected it.